mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -303,6 +322,29 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.1.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
+
+  /** @group setParam */
+  @Since("2.1.0")
+  def setInitialModel(value: Model[_]): this.type = {
```

Comment:
As a follow on, we could eliminate the setter `def setInitialModel(value: Model[_])`. To have better documentation, we could leave the param as abstract in the `HasInitialModel` trait:

````scala
def hasInitialModel: Param[T]
````

Then, when we add this to new models, we implement the param there. So, in KMeansParams:

````scala
/**
 * Param for KMeansModel to use for warm start".
 * @group param
 */
final val hasInitialModel: Param[KMeansModel] = new Param[KMeansModel](this, "initialModel", 
  "A KMeansModel to use for warm start")
````

That way the params are explicit in what type of model is used for initial model and the documentation is more clear.